hazelcast/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -391,8 +394,13 @@ public void init(@Nonnull Context context) {
                             "Local parallelism of " + context.localParallelism() + " was requested for a vertex that "
                                     + "supports only total parallelism of 1. Local parallelism must be 1.");
                 }
-                ownerAddress = context.hazelcastInstance().getPartitionService().getPartition(
-                        StringPartitioningStrategy.getPartitionKey(partitionKey)).getOwner().getAddress();
+                String key = StringPartitioningStrategy.getPartitionKey(partitionKey);
+                int partitionId = context.hazelcastInstance().getPartitionService().getPartition(key).getPartitionId();
+                ownerAddress = context.partitionAssignment().entrySet().stream()
+                        .filter(en -> arrayIndexOf(partitionId, en.getValue()) >= 0)
+                        .findAny()
+                        .map(Entry::getKey)
+                        .orElseThrow(() -> new RuntimeException("Owner partition not assigned to any participating member"));
```

Comment:
We don't expect to enter the exception throwing case here, right? Or I wonder when the program goes into the exception case here

Reply:
No, it should never happen. Every partition should be owned by some processor. If you're curious where will it go look here: https://github.com/hazelcast/hazelcast/blob/0057d5c4c2174ef7bc9aee7ca75b542e4edcf152/hazelcast/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java#L166